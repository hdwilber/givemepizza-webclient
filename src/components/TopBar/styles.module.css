.menu {
	list-style: none;
	padding: 0;
	margin: 0;
}

.menu li {
	display: inline-block;
	margin-right: 1rem;
	cursor: pointer;
	padding: 1rem;
	background-color: #dedede;
}

.menu li:last-child {
	margin-right: inherit;
}

li.active {
	background-color: #8a8a8a;
	color: white;
	font-weight: bold;
}
