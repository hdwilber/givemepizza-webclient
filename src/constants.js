export const Pages = {
  pizzas: 0,
  toppings: 1,
}

