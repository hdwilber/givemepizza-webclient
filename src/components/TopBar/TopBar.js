import React from 'react'
import PropTypes from 'prop-types'
import styles from './styles.module.css'
import { Pages } from '../../constants'

class TopBar extends React.PureComponent {
  render() {
    const { onSelect, current } = this.props
    return (
      <ul className={styles.menu}>
        <li
          className={Pages.pizzas === current ? styles.active: ''}
          onClick={() => onSelect(Pages.pizzas)}
        >
          Pizzas
        </li>
        <li 
          className={Pages.toppings === current ? styles.active: ''}
          onClick={() => onSelect(Pages.toppings)}
        >
          Toppings
        </li>
      </ul>
    )
  }
}
TopBar.propTypes = {
  onSelect: PropTypes.func,
  current: PropTypes.any,
}
export default TopBar

