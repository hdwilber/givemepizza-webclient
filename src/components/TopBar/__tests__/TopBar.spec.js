import React from 'react'
import { mount } from 'enzyme'
import { Pages } from '../../../constants'
import TopBar from '../TopBar'

function setupProps() {
  return {
    onSelect: jest.fn(),
    current: Pages.pizzas,
  }
}

function setup(props) {
  return mount(<TopBar {...props} />)
}

describe('<TopBar />', () => {
  let props
  let wrapper
  beforeEach(() => {
    props = setupProps()
    wrapper = setup(props)
  })

  it('should match snapshot', () => {
    expect(<TopBar {...props} />).toMatchSnapshot()
  })

  it('should call onSelect when user clicks on any item', () => {
    const itemWrapper = wrapper.find('li').at(1)
    itemWrapper.simulate('click')
    expect(props.onSelect).toHaveBeenCalledTimes(1)
    expect(props.onSelect).toHaveBeenCalledWith(Pages.toppings)
  })
  it('should add the correct className when item is the current selected', () => {
    const itemWrapper = wrapper.find('li').at(0)
    expect(itemWrapper.prop('className')).toEqual('active')
  })
})


