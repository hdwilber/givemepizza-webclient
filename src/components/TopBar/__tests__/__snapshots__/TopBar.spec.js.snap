// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> should match snapshot 1`] = `
<TopBar
  current={0}
  onSelect={[MockFunction]}
/>
`;
